scripts/apps/authoring/tests/authoring.spec.js

```diff
@@ -2130,4 +2130,152 @@ describe('send item directive', function() {
             expect(iscope.destination_last.desk).toEqual('123');
             expect(iscope.destination_last.stage).toEqual('456');
         }));
+
+    it('can show send and publish button',
+        inject(function($compile, $rootScope, config) {
+            var scope, elem, iscope;
+            scope = $rootScope.$new();
+            scope.item = {
+                _id: 'foo',
+                type: 'text',
+                state: 'in-progress',
+                task: {
+                    desk: '123',
+                    stage: '456'
+                },
+                _current_version: 1
+            };
+            scope.action = 'edit';
+            elem = $compile('<div sd-send-item data-item="item" data-orig="item" data-mode="authoring" ' +
+                'data-action="action"></div>')(scope);
+            scope.$digest();
+            iscope = elem.isolateScope();
+            expect(iscope.canSendAndPublish()).toBeFalsy();
+            config.ui = {sendAndPublish: 1};
+            expect(iscope.canSendAndPublish()).toBeFalsy();
+            iscope.selectedDesk = {_id: '123'};
+            iscope.selectedStage = {_id: '456'};
+            expect(iscope.canSendAndPublish()).toBeFalsy();
+            iscope.selectedDesk = {_id: '123'};
+            iscope.selectedStage = {_id: '4566'};
+            iscope.itemActions = {publish: 1};
+            expect(iscope.canSendAndPublish()).toBeTruthy();
+            iscope.selectedDesk = {_id: '1234'};
+            iscope.selectedStage = {_id: '456'};
+            expect(iscope.canSendAndPublish()).toBeTruthy();
+        }));
+
+    describe('Send And Publish', function () {
```

Comment:
Since we've already started at this, would you mind removing the spaces between `function` and `(` in this file too (there are a couple of occurrences here).